sanic/asgi.py

```diff
@@ -132,12 +132,20 @@ async def create(
         instance.sanic_app.state.is_started = True
         setattr(instance.transport, "add_task", sanic_app.loop.create_task)
 
-        headers = Header(
-            [
-                (key.decode("latin-1"), value.decode("latin-1"))
-                for key, value in scope.get("headers", [])
-            ]
-        )
+        try:
+            headers = Header(
+                [
+                    (
+                        key.decode("ASCII"),
+                        value.decode(errors="surrogateescape"),
+                    )
+                    for key, value in scope.get("headers", [])
+                ]
+            )
+        except UnicodeDecodeError:
+            raise BadRequest(
+                "Header names can only contain US-ASCII characters"
+            )
```

Comment:
There's no way for anyone who receives this error to know what went wrong. Should we log the header names?

Reply:
Is it safe to directly log the header names from the client? I am worried about some very large header name for flood attack or something like injection. And if we decide to log it later, would it be an error logged by `error_logger`?

Reply:
```suggestion

```

Reply:
You are right. Let's not.